Use dynamic linking for many things; disable jemallocator as it seems
to conflict with libc jemalloc, causing weird test assertion failures

--- Cargo.toml.orig	2020-07-22 18:20:00 UTC
+++ Cargo.toml
@@ -118,12 +118,12 @@ derive_is_enum_variant = "0.1.1"
 leveldb = { git = "https://github.com/timberio/leveldb", optional = true, default-features = false }
 db-key = "0.0.5"
 headers = "0.3"
-rdkafka = { version = "0.24.0", features = ["libz", "ssl", "zstd"], optional = true }
+rdkafka = { version = "0.24.0", features = ["libz", "ssl", "zstd-pkg-config", "dynamic_linking"], optional = true }
 hostname = "0.1.5"
 seahash = { version = "3.0.6", optional = true }
 jemallocator = { version = "0.3.0", optional = true }
 lazy_static = "1.3.0"
-rlua = { git = "https://github.com/kyren/rlua", optional = true }
+rlua = { git = "https://github.com/kyren/rlua", optional = true, default-features = false, features = ["system-lua"] }
 num_cpus = "1.10.0"
 bytesize = { version = "1.0.0", optional = true }
 glob = "0.2.11"
@@ -198,7 +198,7 @@ reqwest = { version = "0.10.6", features = ["json"] }
 
 [features]
 # Default features for *-unknown-linux-gnu and *-apple-darwin
-default = ["sources", "transforms", "sinks", "vendor-all", "unix", "leveldb-plain", "rdkafka-plain"]
+default = ["sources", "transforms", "sinks", "unix", "leveldb-plain", "rdkafka-plain"]
 default-musl = ["sources", "transforms", "sinks", "vendor-all", "unix", "leveldb-cmake", "rdkafka-cmake"]
 # Default features for *-unknown-linux-* which make use of `cmake` for dependencies
 default-cmake = ["sources", "transforms", "sinks", "vendor-all", "unix", "leveldb-cmake", "rdkafka-cmake"]
@@ -207,7 +207,7 @@ default-cmake = ["sources", "transforms", "sinks", "ve
 default-msvc = ["sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "leveldb-cmake", "rdkafka-cmake"]
 
 # Enables features that work only on systems providing `cfg(unix)`
-unix = ["jemallocator"]
+unix = []
 # These are **very** useful on Cross compilations!
 vendor-all = ["vendor-sasl", "vendor-openssl", "vendor-libz"]
 vendor-sasl = ["rdkafka/gssapi-vendored"]
